src/components/modal/index.tsx

```diff
@@ -0,0 +1,100 @@
+
+import { TextField } from '@mui/material';
+import { useContext, useState, useEffect } from 'react';
+import { Modal } from 'react-bootstrap';
+import Button from '@mui/material/Button';
+import { IRepositoryProps } from '../../types/repositories';
+import { UserProfileContext } from '../../context/user';
+import { getProfileUser } from '../../service';
+import { IProfileUserProps } from '../../types/profileUser';
+
+export const ModalComponent = ({children}: any) => {
```

Comment:
Set the type of the ModalComponent's props to an interface instead of use the `any` type.
If here you use the children only you can do something like this:
```tsx
interface ModalComponentProps {
   children: React.ReactNode
}

```
then

```tsx
export const ModalComponent = ({children}: ModalComponentProps) => { /* ... */}
```